### PR TITLE
feat: added option toget users who have receive marketing emails as true

### LINKED ISSUE #158: [FEAT] :sparkles: get all users which who have added option to receive marketing emails
### Description

add new query to fetch all users route `receiveMarketingEmails`.
if this is true than only fetch users in which `receiveMarketingEmails` is set to `true`

example:- `GET /api/admin/users?receiveMarketingEmails=true`

### Anything else?

start working on this only when #157 is closed

### PR DESCRIPTION
**Pull request by shikhar13012001**
### Description
added option to get users who have receive marketing emails as true

### Checklist:

-   [x] This pull request follow our contribution guidelines
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (postman and wiki)
-   [ ] I have written tests for the code
-   [ ] I have updated .env.example file for new .env variables

### What is current behavior?

Closes #158

### CODE REVIEW

File: src/controllers/user.controller.ts [hunk 1]
```diff
@@ -29,15 +29,16 @@ export const getAllUsersHandler = async (
...
 	try {
-		let { page, size } = req.query;
+		let { page, size, receiveMarketingEmails } = req.query;
 
 		if (!page) page = "1";
 		if (!size) size = "10";
 
 		const limit = +size;
 		const skip = (+page - 1) * limit;
+		const query = receiveMarketingEmails ? { receiveMarketingEmails } : {};
```
Comment (aayushchugh) [inline src/controllers/user.controller.ts]: The case of this field in the database is changed, please look at that

**Review by aayushchugh — CHANGES_REQUESTED** [review]

File: src/controllers/user.controller.ts (hunk 1 above)
Comment (shikhar13012001) [reply src/controllers/user.controller.ts]: Thanks for pointing it , I just overlooked it , sorry. I will make the changes.

**Comment by shikhar13012001** [comment]
@aayushchugh I have made requested changes as required, please review the PR and let me know if you need some changes.
Thanks!

**Review by aayushchugh — APPROVED** [review]
Looks good

**Comment by aayushchugh** [comment]
@all-contributors please add @shikhar13012001 for code

**Pull request merged**